models/issue_dependency.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2017 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	"time"
+)
+
+// IssueDependency is connection request for receiving issue notification.
+type IssueDependency struct {
+	ID           int64     `xorm:"pk autoincr"`
+	UserID       int64     `xorm:"UNIQUE(watch) NOT NULL"`
+	IssueID      int64     `xorm:"UNIQUE(watch) NOT NULL"`
+	DependencyID int64     `xorm:"UNIQUE(watch) NOT NULL"`
+	Created      time.Time `xorm:"-"`
+	CreatedUnix  int64     `xorm:"INDEX created"`
+	Updated      time.Time `xorm:"-"`
+	UpdatedUnix  int64     `xorm:"updated"`
+}
+
+// Define Dependency Type Constants
+const (
+	DependencyTypeBlockedBy int64 = 1
+	DependencyTypeBlocking  int64 = 2
+)
+
+// CreateIssueDependency creates a new dependency for an issue
+func CreateIssueDependency(user *User, issue, dep *Issue) (exists bool, err error) {
+	sess := x.NewSession()
+
+	// Check if it aleready exists
+	exists, err = issueDepExists(x, issue.ID, dep.ID)
+	if err != nil {
+		return exists, err
+	}
+
+	// If it not exists, create it, otherwise show an error message
+	if !exists {
+		newIssueDependency := &IssueDependency{
+			UserID:       user.ID,
+			IssueID:      issue.ID,
+			DependencyID: dep.ID,
+		}
+
+		if _, err := x.Insert(newIssueDependency); err != nil {
+			return exists, err
+		}
+
+		// Add comment referencing the new dependency
+		_, err = createIssueDependencyComment(sess, user, issue, dep, true)
+
+		if err != nil {
+			return exists, err
+		}
+
+		// Create a new comment for the dependent issue
+		_, err = createIssueDependencyComment(sess, user, dep, issue, true)
+
+		if err != nil {
+			return exists, err
+		}
+	}
+	return exists, nil
+}
+
+// RemoveIssueDependency removes a dependency from an issue
+func RemoveIssueDependency(user *User, issue *Issue, dep *Issue, depType int64) (err error) {
+	sess := x.NewSession()
+
+	// Check if it exists
+	exists, err := issueDepExists(x, issue.ID, dep.ID)
```

Comment:
You do check where you switch issue for dep, but later you are deleting in given order. What it should do?

Reply:
It checks if the dependency already exists as you send it to the backend, but it also checks the other way around to prevent to issues blocking each other. This would result in none of them could be closed.

Reply:
ok